Design.md

```diff
@@ -32,71 +32,73 @@ Here are the IELE design decisions described from the perspective of differences
 IELE is a register-based bytecode language, unlike EVM, which is a stack-based bytecode. As a result, the following alterations have been made:
 
 * Each instruction in the bytecode takes register operands representing the arguments and result of the instruction.
-* A REGISTERS instruction is added to the language. The contract is malformed if REGISTERS is not placed at the beginning of the contract. It sets the bit width of a register operand in the bytecode. If no REGISTERS instruction exists, the bit width defaults to 5 bits
+* There is an unlimited number of registers available and they can be referred to by user-defined names, similar to the virtual registers of LLVM. The more registers a contract uses, the more gas it consumes.
 * Because the stack no longer exists, it is no longer possible for the VM to throw an exception due to stack underflow or overflow.
-* Because the stack no longer exists, the POP, DUP, SWAP, and PUSH instructions no longer exist. They are replaced with a LOADPOS, LOADNEG, and MOVE instruction. LOADPOS and LOADNEG perform the function of loading an immediate value into a register (see section on arbitrary-precision integers for more details), and MOVE copies a value from one register to another.
-* Registers are divided into two categories: global registers and local registers. Half of the declared registers are global; the other half are local.
+* Because the stack no longer exists, the POP, DUP, SWAP, and PUSH instructions no longer exist. They are replaced with an assignment instruction that can be used to load a constant value into a register (see section on arbitrary-precision integers for more details), and/or copy a value from one register to another.
+* Registers are divided into two categories: global registers and local registers. Local registers are visible only within the function in which they are used, while global registers are visible across all functions of a given contract. Similar to LLVM, global registers names are prefixed by `@`, while local register names are prefixed by `%`.
+* Local registers have a lifetime equal to the lifetime of a function call to the function that contains them, while global registers have a lifetime that starts with the creation of an account with an instance of the contract that contains them and ends with the destruction of said account.
```

Comment:
This sounds like global register values are preserved between transactions, like storage.

Reply:
Actually, by reading the current semantics it is not clear to me what happens to global registers across account calls. But if they are not preserved, then what's the point of even having them?
Any comments @dwightguth ?